auxx/src/Lang/DisplayError.hs

```diff
@@ -98,7 +98,7 @@ renderFullLine str = renderLine 0 (length str) str
 ppParseError :: ParseError -> Doc
 ppParseError (ParseError str (Report {..})) =
       "Parse error at" <+> text (show span)
-  <$> "Unexpected" <+> text unconsumedDesc <> ", expected"
+  <$> "Unexpected " <+> text unconsumedDesc <+> ", expected"
```

Comment:
And here. Why the space after `Unexpected`? `<+>` concatenates with adding a space in-between, so in the end that part of the message will have two spaces. And why add space before the comma?

Reply:
Oh, I got you. I thought `<>` is the same as `<+>` here.

Reply:
No, they aren't.
`<$>` and `<+>` here are special Doc combinators.
https://hackage.haskell.org/package/ansi-wl-pprint-0.6.8.1/docs/Text-PrettyPrint-ANSI-Leijen.html#v:-60--43--62-